airflow/executors/celery_kubernetes_executor.py

```diff
@@ -0,0 +1,157 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Dict, Optional, Set, Union
+
+from airflow.configuration import conf
+from airflow.executors.base_executor import CommandType, EventBufferValueType, QueuedTaskInstanceType
+from airflow.executors.celery_executor import CeleryExecutor
+from airflow.executors.kubernetes_executor import KubernetesExecutor
+from airflow.models.taskinstance import SimpleTaskInstance, TaskInstance, TaskInstanceKey
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+
+class CeleryKubernetesExecutor(LoggingMixin):
```

Comment:
Ok I have to do this 😄 I was thinking about this yesterday but was not sure... Do you think that this name is good? On one hand, it's explicit on the other... I can imagine questions like "Is this for Celery on Kubernetes?", "Can I use CeleryExecutor on kubernetes? Or do I have to use this one?"

Do I have any better ideas? `CeleryAndKubernetesExecutor`? Is this better? I don't know just want to hear what others think 🤔

Reply:
Good point about the naming.  i like it, `CeleryAndKubernetesExecutor` which indicates `celery` and `kubernetes` are both used in this class.

Reply:
No strong opinion on the naming of the class, I am fine with either since we have the first line of the docstring explaining it :)

Reply:
@turbaszek @pingzh Maybe we can name it the "MultiExecutor" or "JoinedExecutor"? I could see a case where in the future we route even more executors into this (like Dask)

Reply:
@dimberman thanks for the suggestion. I think we can go further to have a `BaseComposeddExecutor`, which serves as the base class for any composed executor. The only two methods for the subclass to override is the `__init__` and also the `_route` method.

However, I am thinking of whether we over-optimize it in this case. I don't want to have a too generic name at the beginning. We can come back to refactor the logic when we have another use case like this.